sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -57,20 +58,34 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
   internal var lifecycleState: LifecycleState = LifecycleState.STATE_STOPPED
 
   constructor(
-    renderer: MapboxRenderer,
+    renderer: MapboxRendererInterface,
     mapInitOptions: MapInitOptions,
   ) {
     this.renderer = renderer
     this.mapInitOptions = mapInitOptions
     AssetManagerProvider().initialize(mapInitOptions.context.assets)
-    this.nativeMap = MapProvider.getNativeMap(
-      mapInitOptions,
-      renderer,
-    )
+    // create Map in Kotlin if using Android renderer
+    if (renderer is MapboxRenderer) {
+      this.nativeMap = MapProvider.getNativeMap(
+        mapInitOptions,
+        renderer
+      )
+    } else {
+      // create Map in C++ if using C++ renderer and return it to Kotlin
+      if (renderer is MapboxNativeSurfaceRenderer) {
+        this.nativeMap = NativeMapImpl(
```

Comment:
Since we already have `MapProvider.getNativeMap(` - maybe move native map there as well? 
Also, since previous method is already called `NativeMap` - maybe name new one somehow differently>